core/src/main/scala/org/apache/spark/metrics/sink/GraphiteSink.scala

```diff
@@ -69,7 +69,7 @@ private[spark] class GraphiteSink(val property: Properties, val registry: Metric
 
   val graphite = propertyToOption(GRAPHITE_KEY_PROTOCOL).map(_.toLowerCase(Locale.ROOT)) match {
     case Some("udp") => new GraphiteUDP(new InetSocketAddress(host, port))
-    case Some("tcp") | None => new Graphite(new InetSocketAddress(host, port))
+    case Some("tcp") | None => new Graphite(host, port)
```

Comment:
Do we need to change here also?

Reply:
@jerryshao  Do you mean change Graphite constructor for case "udp" ?

Reply:
Yes, that's what I mean, I'm not sure if it is required since I'm not familiar with this Graphite sink.

Reply:
Looks like there is a constructor accepting hostname directly for UDP. I'll add that as well.